src/main/java/org/sefglobal/scholarx/service/ProgramService.java

```diff
@@ -185,6 +162,43 @@ public Program updateState(long id) throws ResourceNotFoundException {
         return programRepository.save(program.get());
     }
 
+    public void sendMenteeApplicationEmails(long id, Optional<Program> program) {
+        List<Mentor> mentors = mentorRepository.findAllByProgramId(id);
+
+        String message;
+        for (Mentor mentor : mentors) {
+            message = "You have been " + mentor.getState().name().toLowerCase();
+            emailService.sendEmail(mentor.getProfile().getEmail(), program.get().getTitle(), message);
+        }
+    }
+
+    public void sendMenteeSelectionEmails(long id, Optional<Program> program) {
+        List<Mentor> approvedMentors = mentorRepository.findAllByProgramIdAndState(id, EnrolmentState.APPROVED);
+
+        String message = "You can approve or reject your mentees by visiting the dashboard";
+        for (Mentor mentor : approvedMentors) {
+            emailService.sendEmail(mentor.getProfile().getEmail(), program.get().getTitle(), message);
+        }
+    }
+
+    public void sendOnGoingEmails(long id, Optional<Program> program) {
+        List<Mentor> approvedMentors = mentorRepository.findAllByProgramIdAndState(id, EnrolmentState.APPROVED);
+
+        String message = "You can check your mentees by visiting the dashboard";
+        for (Mentor mentor : approvedMentors) {
+            emailService.sendEmail(mentor.getProfile().getEmail(), program.get().getTitle(), message);
+        }
+    }
+
+    public void sendMentorConfirmationEmails(long id, Optional<Program> program) {
+        List<Mentee> mentees = menteeRepository.findAllByProgramId(id);
+
+        String message = "You can check your mentor by visiting the dashboard";
+        for (Mentee mentee : mentees) {
+            emailService.sendEmail(mentee.getProfile().getEmail(), program.get().getTitle(), message);
+        }
+    }
+
```

Comment:
Shall we move these functions into the util dir?
You can import them back to use them here, It's not a good practice to keep these functions in the service layer itself.

Reply:
oh I see, sure I will do that 🙂️

Reply:
@DilshanNaveen you still working on this?

Reply:
@Gravewalker666 I'm extremely sorry for the delay, I will send this update as soon as possible

Reply:
Nice, it's alright to get a lil break from the task
We all have other commitments after all 😄 
No worries at all dude
Don't forget to keep us updated next time

Reply:
sure, thanks a lot

Reply:
@Gravewalker666, I updated the pull request with the changes requested, can you check?

Reply:
Yeah, sure

Reply:
The code looks fine to me. 
@anjula-sack @Piumal1999 can you check the logical part of the email functions?? 
I have a few doubts and a bit foggy about how it previously was.